x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -38,20 +40,53 @@ const DENY_ALL_PLUGINS = {
 export const checkAccess = async ({
   config,
   security,
+  spaces,
   request,
   log,
 }: ICheckAccess): Promise<IAccess> => {
+  const isRbacEnabled = security?.authz.mode.useRbacForRequest(request) ?? false;
+
+  // We can only retrieve the active space when either:
+  // 1) security is enabled, and the request has already been authenticated
+  // 2) security is disabled
+  const attemptSpaceRetrieval = !isRbacEnabled || request.auth.isAuthenticated;
+
+  // If we can't retrieve the current space, then assume the feature is available
+  let allowedAtSpace = false;
+
+  if (!spaces) {
+    allowedAtSpace = true;
+  }
+
+  if (spaces && attemptSpaceRetrieval) {
+    try {
+      const space = await spaces.spacesService.getActiveSpace(request);
+      allowedAtSpace = !space.disabledFeatures?.includes('enterpriseSearch') ?? true;
```

Comment:
Is it a realistic scenario that the `disabledFeatures` key is completely missing? If it's not likely, maybe we could simplify this down to

```suggestion
      allowedAtSpace = !space.disabledFeatures?.includes('enterpriseSearch');
```

which returns true in any case if `disabledFeatures` is undefined 🤷 

That way we don't have to cover the `?? true` conditional which is the only thing I have left locally as an uncovered line/branch 😄

Reply:
This isn't something that should be possible (we have a Saved Object migration in place to add it if it's missing), but we used to encounter places where it was missing anyway. I was trying to be extra cautious, but in reality it's probably unnecessary. I'll simplify 👍